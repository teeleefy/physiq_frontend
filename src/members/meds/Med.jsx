import {
    Card,
    CardBody,
    CardText,
    ListGroupItem
  } from "reactstrap";
  import { NavLink } from "react-router-dom";
// import '../styles/Allergy.css'


function Med({med, memberId}){

    return(
        <>
            <ListGroupItem>
            <Card className="Med-card">
                <CardBody>
                    <CardText>
                        <p><b>Medication Name:</b> <NavLink className="Med-name" to={`/member/${memberId}/meds/${med.id}`} key={med.id}>
                            {med.name}
                        </NavLink></p>
                        <p><b>Indication:</b> {med.indication}</p>
                        <p><b>Dose:</b> {med.dose}</p>
                        <p><b>Start Date:</b> {med.startDate}</p>
                        <p><b>End Date:</b> {med.endDate}</p>
                        <p><b>Prescribed By:</b> {med.prescriber}</p>
                        <p><b>Notes:</b> {med.notes}</p>
                    </CardText>
                </CardBody>
            </Card>
            </ListGroupItem>
        </>
     )
}

export default Med;