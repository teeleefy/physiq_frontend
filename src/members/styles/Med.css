.Med-card {
    height: fit-content;
    margin-left: 0.5rem;
    margin-bottom: 0.5rem;
}

.Med-label {
    font-size: 1.2rem;
}

.Med-h1 {
    font-size: 2rem;
    font-weight: 600;
}

.Meds-navlink {
    text-decoration: none;
    color: white;
}

.Med-date-input {
    width: 20vw;
}

.Med-select-input {
    width: 45vw;
}