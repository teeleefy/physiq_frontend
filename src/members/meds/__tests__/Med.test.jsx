import { renderWithRouter, screen } from '../../../../__tests__/test-tools';
import Med from '../Med'

let med={name: "Lisinopril", indication: "High blood pressure", dose: "25mg twice a day", startDate: "2023-12-20", endDate: "2024-12-20", prescriber: "Dr. Jones", notes: "Take with food."}
let memberId=1000;

describe('Med', () => {
  it('renders the Med component', () => {
    renderWithRouter(<Med med={med} memberId={memberId}/>)
    
    screen.debug(); // prints out the jsx in the Med component unto the command line
  })
})

it("matches snapshot", function() {
    const { asFragment } = renderWithRouter(<Med med={med} memberId={memberId}/>);
    expect(asFragment()).toMatchSnapshot();
  });