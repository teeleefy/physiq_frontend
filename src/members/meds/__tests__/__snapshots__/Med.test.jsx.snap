// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="list-group-item"
  >
    <div
      class="Med-card card"
    >
      <div
        class="card-body"
      >
        <p
          class="card-text"
        />
        <p>
          <b>
            Medication Name:
          </b>
           
          <a
            class="Med-name"
            href="/member/1000/meds/undefined"
          >
            Lisinopril
          </a>
        </p>
        <p>
          <b>
            Indication:
          </b>
           High blood pressure
        </p>
        <p>
          <b>
            Dose:
          </b>
           25mg twice a day
        </p>
        <p>
          <b>
            Start Date:
          </b>
           2023-12-20
        </p>
        <p>
          <b>
            End Date:
          </b>
           2024-12-20
        </p>
        <p>
          <b>
            Prescribed By:
          </b>
           Dr. Jones
        </p>
        <p>
          <b>
            Notes:
          </b>
           Take with food.
        </p>
        <p />
      </div>
    </div>
  </li>
</DocumentFragment>
`;
